backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java

```diff
@@ -1211,6 +1216,146 @@ public MessageHeader createMessageHeader(
     messageHeader
         .getMeta()
         .addTag(PROCESSING_ID_SYSTEM, processingId, PROCESSING_ID_DISPLAY.get(processingId));
+    messageHeader.addFocus(new Reference(mainResourceUrl));
+
+    if (!StringUtils.isBlank(organizationUrl)) {
+      messageHeader.getSender().setReferenceElement(new StringType(organizationUrl));
+    }
+    if (!StringUtils.isBlank(provenanceFullUrl)) {
+      messageHeader.addFocus(new Reference(provenanceFullUrl));
+    }
     return messageHeader;
   }
+
+  public DiagnosticReport convertToDiagnosticReport(
+      ConditionAgnosticConvertToDiagnosticReportProps props) {
+    DiagnosticReport diagnosticReport = new DiagnosticReport();
+    var categoryCodeableConcept = new CodeableConcept();
+    var categoryCoding = categoryCodeableConcept.addCoding();
+    categoryCoding.setSystem(DIAGNOSTIC_CODE_SYSTEM);
+    categoryCoding.setCode("LAB");
+    diagnosticReport.setCategory(List.of(categoryCodeableConcept));
+
+    var diagnosticCodeableConcept = new CodeableConcept();
+    var diagnosticCoding = diagnosticCodeableConcept.addCoding();
+
+    diagnosticCoding.setSystem(LOINC_CODE_SYSTEM).setCode(props.getTestPerformedCode());
+    diagnosticReport.setCode(diagnosticCodeableConcept);
+
+    var effectiveDateTime = DateTimeUtils.convertToZonedDateTime(props.getTestEffectiveDate());
+    diagnosticReport.setEffective(convertToDateTimeType(effectiveDateTime));
+
+    String diagnosticReportId = uuidGenerator.randomUUID().toString();
+    diagnosticReport.setId(diagnosticReportId);
+
+    return diagnosticReport;
+  }
+
+  public Observation convertToObservation(ConditionAgnosticConvertToObservationProps props) {
+    Observation observation = new Observation();
+    setStatus(observation, props.getCorrectionStatus());
+
+    var labCodeableConcept = new CodeableConcept();
+    var labCodeableCoding = labCodeableConcept.addCoding();
+    labCodeableCoding.setSystem(OBSERVATION_CATEGORY_CODE_SYSTEM);
+    labCodeableCoding.setCode("laboratory");
+    observation.setCategory(List.of(labCodeableConcept));
+
+    var observationCodeableConcept = new CodeableConcept();
+    var observationCoding = observationCodeableConcept.addCoding();
+    observationCoding.setSystem(LOINC_CODE_SYSTEM).setCode(props.getTestPerformedCode());
+    observation.setCode(observationCodeableConcept);
+
+    addSNOMEDValue(props.getResultValue(), observation, "");
```

Comment:
let me know if passing in an empty string / gating it in the inner method to not include the value in the bundle is bad practice. Happy to refactor using some sort of null check, overload, etc.

Reply:
I think it should be better to just add null rather than empty string here to represent nothing. I see that you are already using [`StringUtils.isBlank`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#isBlank-java.lang.CharSequence-) already so it will already handle that null case.

<img width="535" alt="image" src="https://github.com/CDCgov/prime-simplereport/assets/10108172/cb672c9e-8d10-4e62-9277-7f2e9f56237d">